Create CA certificate for JGroups encryption (#37044)
keycloak/keycloak: docs/guides/server/caching.adoc

The truststore contains the valid certificates to accept connection from, and it can be configured with `cache-embedded-mtls-trust-store-file` (path to the truststore), and `cache-embedded-mtls-trust-store-password` (password to decrypt it).
To restrict unauthorized access, use a self-signed certificate for each {project_name} deployment.

[NOTE]
====
**Zero Configuration Encryption**

{project_name} offers a zero-configuration approach to encrypting network communication between nodes.

> This feature automatically generates self-signed certificates, eliminating the need for manual certificate creation and management.

Reviewer: You should mention when it's set for the main HTTP server, it's inherited, right?

Reviewer 2: This keys/certificate is just for the JGroups internal cache communication.

The truststore contains the valid certificates to accept connection from, and it can be configured with `cache-embedded-mtls-trust-store-file` (path to the truststore), and `cache-embedded-mtls-trust-store-password` (password to decrypt it).
To restrict unauthorized access, use a self-signed certificate for each {project_name} deployment.

[NOTE]
====
**Zero Configuration Encryption**

{project_name} offers a zero-configuration approach to encrypting network communication between nodes.
This feature automatically generates self-signed certificates, eliminating the need for manual certificate creation and management.

> The generated certificate and associated keys are securely stored within the database of each {project_name} instance.

Reviewer 2: ```suggestion
The generated certificate and associated keys are stored within the database of each {project_name} instance.
```

To enable zero-configuration TLS encryption, set the `cache-embedded-mtls-enabled` option to true.
It's important to note that this disables any custom certificate configurations specified using other cache-embedded-mtls-* options.

> Therefore, for zero-configuration encryption, you should not set any other cache-embedded-mtls-* options.

Author: sure!

> It's important to note that this disables any custom certificate configurations specified using other cache-embedded-mtls-* options.
> Therefore, for zero-configuration encryption, you should not set any other cache-embedded-mtls-* options.

Reviewer 2: I now realize that we don't have a new option for zero config encryption. So the text should read a bit different: 

```suggestion
No other `cache-embedded-mtls-*` must be set to enable the zero-configuration mode.
```
